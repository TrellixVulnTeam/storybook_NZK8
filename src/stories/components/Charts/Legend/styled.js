// @flow
import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

export const LegendItem = styled.div`
  display: flex;
  margin: 6px 4px;
  align-items: center;
  > span {
    margin-right: 5px;
  }
`;

export const Marker = styled.span`
  width: 15px;
  height: 15px;
  border-radius: 4px;
  background-color: ${p => p.bgColor};
`;
export const Label = styled.span`
  width: 100px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;
export const Value = styled.span`
  margin-left: 15px;
  color: ${colors.steel};
  font-size: 14px;
`;
