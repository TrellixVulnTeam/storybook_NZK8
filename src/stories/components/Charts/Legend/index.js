// @flow
/* eslint-disable import/prefer-default-export */
import React from 'react';
import { LegendItem, Marker, Label, Value } from './styled';
/**
 * Renders Legend items
 * @param {Object} props
 * @return {Node}
 */
export const renderLegend = (props: Object) => {
  const { payload } = props;
  return payload.map((entry, index) => (
    // eslint-disable-next-line react/no-array-index-key
    <LegendItem key={`legend-item-${index}`}>
      <Marker bgColor={entry.color} />
      <Label>{entry.value}</Label>
      <Value>{entry.payload.value}%</Value>
    </LegendItem>
  ));
};
