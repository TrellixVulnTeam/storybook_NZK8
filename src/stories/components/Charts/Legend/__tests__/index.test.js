import { renderLegend } from 'components/Charts/Legend/index';

describe('<renderLegend />', () => {
  const props = {
    payload: [
      {
        color: 'red',
        value: 0,
        payload: { value: 0 },
      },
      {
        color: 'blue',
        value: 1,
        payload: { value: 1 },
      },
      {
        value: 2,
        payload: { value: 2 },
      },
    ],
  };

  it('should render correctly with props', () => {
    const sut = renderLegend(props);

    expect(sut).toMatchSnapshot();
  });
});
