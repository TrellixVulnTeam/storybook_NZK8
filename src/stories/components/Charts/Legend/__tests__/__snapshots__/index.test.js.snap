// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<renderLegend /> should render correctly with props 1`] = `
Array [
  <styled__LegendItem>
    <styled__Marker
      bgColor="red"
    />
    <styled__Label>
      0
    </styled__Label>
    <styled__Value>
      0
      %
    </styled__Value>
  </styled__LegendItem>,
  <styled__LegendItem>
    <styled__Marker
      bgColor="blue"
    />
    <styled__Label>
      1
    </styled__Label>
    <styled__Value>
      1
      %
    </styled__Value>
  </styled__LegendItem>,
  <styled__LegendItem>
    <styled__Marker />
    <styled__Label>
      2
    </styled__Label>
    <styled__Value>
      2
      %
    </styled__Value>
  </styled__LegendItem>,
]
`;
